### ISSUE TITLE
Session Loss in WebForms

### ISSUE THREAD

**Issue by ichibanchef** [couchbaselabs/couchbase-aspnet#71]
Something's not stirring the Kool-Aid with the session state provider in a WebForms environment. It seems as if a portion of the session data is either never being stored or is immediately timing out. I'm assuming this because we typically see two entries in the console for each session, but with the new provider we're only seeing one. The observed behavior is that sessions appear to time out immediately after they are started.

I haven't had time to try and narrow down a cause for this just yet. I've used both the "section" and "inline" bootstrap strategies with a range of different config options; always ending with the same result.

**Comment by jeffrymorris** [comment]
@ichibanchef -

Thanks for reporting, I'll look deeper into it. The 3.0 provider doesn't split the header and body of the session state like the previous (1.X and 2.X) providers did, so seeing a single value is expected. However, the instant timeout seems like the bug. Can you post an configuration and usage example?

Jeff

**Comment by jeffrymorris** [comment]
@ichibanchef -

This commit should fix the issue: https://github.com/couchbaselabs/couchbase-aspnet/commit/b46f0086a861937708218546446ec97a6bc9685c

**Comment by ichibanchef** [comment]
Jeff - I've pulled down 3.0.0-beta2, but something's still not right with the session provider. The symptoms are similar to the original submission, but with one exception. Initial login attempts immediately time out when redirecting from the login page on our test site as mentioned before. When observing with the debugger, the Session.Timeout value is always 0. This is reproducible 100% of the time.

_However_... If I clean and build the test website project things get weird. The initial login attempt will timeout immediately, but all subsequent login attempts with the same browser instance will succeed and the timeout issue does not occur. _BUT_, If I close the browser instance and open a new one, or open a different browser, and attempt a login, the persistent timeout after redirect resumes. So, the first session created after a site clean and build will work after initial failure, but any new sessions created consistently time out.

I've attached the relevant bits of the web.config. It's very much a "no frills" config. The test site runs on IIS 10 on my dev box on the standard port 80, in a virtual directory. Doing an iisreset does not appear to affect the symptoms.

[web.config.txt](https://github.com/couchbaselabs/couchbase-aspnet/files/1767659/web.config.txt)

**Comment by jeffrymorris** [comment]
Hi @ichibanchef -

Thanks for the detailed steps, i'll try to reproduce. If you can enable debugging and take a capture while doing the steps described, it might also help identify the problem.

I re-opened the issue until we get a resolution; if it ends up being something different, i'll open another issue for it.

-Jeff

**Comment by jeffrymorris** [comment]
@ichibanchef -

I tried to replicate this behavior and I couldn't, perhaps you can supply a simple example project?

Thanks,

Jeff

**Comment by jeffrymorris** [comment]
Hi @ichibanchef -

Following up on this issue, are you still encountering it? Can you provide an example project that illustrates the behavior?
Thanks,

-Jeff

**Comment by jeffrymorris** [comment]
@ichibanchef if no response by EOW, i'll assume this is no longer an issue and close the ticket. Please respond if you would like to keep it open!

Thanks,

Jeff

**Comment by ichibanchef** [comment]
Jeff,

Sorry I haven't been back. I got pulled onto another project and have not had a chance to revisit this again yet. I will see what I can do about getting a test project going to try and replicate the issue. We replaced the DLLs and updated configs in a site that had the previous flavor of the aspnet that was working with couchbase 4. No other changes were made to our dev environment beyond that. We don't seem to be having any issues with caching things with the netclient, just issues with session on the aspnet side.

Ash
